### DISCUSSION TITLE
2.0.0 Beta 1

### DISCUSSION THREAD

**Discussion by x1unix** [x1unix/go-playground#323] [Announcements]
## Description

This is the first public preview of the upcoming 2.0.0 release, which includes many major changes.

## Where To Try?

The unstable version is available at [staging.goplay.tools](https://staging.goplay.tools).

> [!WARNING]  
> This is an unstable version. Please feel free to report any found bugs.

### Multi File Support

This release brings one of the most requested features - multiple files support.

![image](https://github.com/x1unix/go-playground/assets/9203548/b677d507-b926-4b3d-953d-ab57e2ae7ae6)

### Third-Party Packages in WebAssembly

We added support for third-party packages for WebAssembly Environment.

![image](https://github.com/x1unix/go-playground/assets/9203548/9f783c07-380c-4261-8915-91b04ecee2c3)

### Improved Examples

Examples were moved to a more recognizable place and the UI of examples was also improved.

![image](https://github.com/x1unix/go-playground/assets/9203548/7f28c74a-b1ac-4c7b-adc3-eb79f742b214)

### Refactoring

This release also brings a lot of changes that happened under the hood.

* Project migrated from create-react-app to [Vite](https://vitejs.dev/).
* Replaced old Monaco component with [@monaco-editor/react](https://www.npmjs.com/package/@monaco-editor/react).
* Lots of refactoring on frontend and backend sides.

<hr /><em>This discussion was created from the release <a href='https://github.com/x1unix/go-playground/releases/tag/v2.0.0-beta.1'>2.0.0 Beta 1</a>.</em>

**Comment by vuon9** [comment]
This is awesome! 👏
